grails-web-url-mappings/src/main/groovy/org/grails/web/mapping/DefaultUrlMappingInfo.java

```diff
@@ -212,6 +215,13 @@ public String getId() {
         return evaluateNameForValue(id);
     }
 
+    /**
+     * @deprecated
+     * This method will be removed in a future grails version since the associated g:submitAction is being removed.
+     * Grails will no longer support redirecting to a different action name by adding a parameter with the prefix
+     * '_action'
+     */
+    @Deprecated(since = "7.0.0")
     private String checkDispatchAction(HttpServletRequest request) {
```

Comment:
Is it necessary to deprecate a private method?

Reply:
I guess we could use a todo instead.  I would prefer the annotation because it lets us know when it was deprecated so we know when to remove it.  Also, it lets us know via compiler checks

Reply:
I think that is useful.   It seems like there is a history of not removing things as planned.  Annotating makes it clearer.

Reply:
would it be beneficial to say use g: formActionSubmit instead in the warning message?

Reply:
It's possible someone is relying on this behavior to redirect for a reason we don't know about.  To my knowledge, the only "framework" reason is that tag ... so I guess.  Do you have a proposed wording?

Reply:
Should we also add `forRemoval = true` attribute?

Reply:
I added the attribute, @codeconsole do you have a suggestion on the warning message?

Reply:
How about just add it at the end?  The tag is the only place this is used, right?

```java
"Dispatch Action [%s] detected; Dispatch Actions will be removed in a future version of Grails. Use g: formActionSubmit instead."
```